src/ACEtk/block/details/BaseDistributionData/src/generateBlocks.hpp

```diff
@@ -0,0 +1,13 @@
+void generateBlocks() {
+
+  std::size_t nr = static_cast< std::size_t >( this->XSS( 1 ) );
+  std::size_t ne = static_cast< std::size_t >( this->XSS( 1 + 2 * nr + 1 ) );
```

Comment:
[suggestion - complexity]
The entry in the ACE format manual is simply `2 + 2*nr`.

Reply:
For my own comprehension I often do things like this.

One value for NR, followed by two times NR values, followed by the NE value. The compiler should simplify this to 2 + 2*nr easily.